### ISSUE TITLE
incorrect handling of if-modified-since header

### ISSUE THREAD

**Issue by randypuro** [dustyburwell/connect-file-cache#1]
cache.js line 80. The in memory mtime includes milliseconds; whereas the client's header string does not making this comparison almost always presume that the file has been modified even if it hasn't.

**Comment by TrevorBurnham** [comment]
Could you provide a test case? The current integration test passes:

```
exports['Conditional GET can yield a 304 "Not Modified" response'] = (test) ->
  request 'http://localhost:3688/popeye-zen.txt', (err, res, body) ->
    mtime = res.headers['last-modified']
    options =
      url: 'http://localhost:3688/popeye-zen.txt'
      headers: {'If-Modified-Since': mtime}
    request options, (err, res, body) ->
      test.equals body, ''
      test.equals res.statusCode, 304
      test.done()
```

The `If-Modified-Since` header that gets sent (on my machine) is `Fri, 23 Sep 2011 23:33:21 GMT`, and that successfully yields a 304 response.

**Comment by randypuro** [comment]
Yes, sorry for the delay. It's basically a problem when populating the cache using the set method, rather than from the file system (eg. as done in connect-assets). The mtime stored in the cache will have millis, whereas If-Modified-Since will not. 

```
exports['Conditional GET yields a 304 "Not Modified" response for mtime with millis'] = (test) ->
  cache.set('newroute.txt', 'some text', mtime: new Date)

  request 'http://localhost:3688/newroute.txt', (err, res, body) ->
    test.equals body, 'some text'
    test.equals res.statusCode, 200
    mtime = res.headers['last-modified']

    options =
      url: 'http://localhost:3688/newroute.txt'
      headers: {'If-Modified-Since': mtime}
    request options, (err, res, body) ->
      test.equals body, ''
      test.equals res.statusCode, 304
      test.done()
```

**Comment by TrevorBurnham** [comment]
Thanks very much for this, and sorry for the delay in fixing. It's resolved in 0.2.3 (and in connect-assets 2.1.1).
